controllers/remotesecret_controller.go

```diff
@@ -514,19 +561,15 @@ func (f *remoteSecretLinksFinalizer) Finalize(ctx context.Context, obj client.Ob
 
 	for i := range remoteSecret.Status.Targets {
 		ts := remoteSecret.Status.Targets[i]
-		dep := bindings.DependentsHandler[*api.RemoteSecret]{
-			Target: &namespacetarget.NamespaceTarget{
-				Client:       f.client,
-				TargetKey:    key,
-				SecretSpec:   &remoteSecret.Spec.Secret,
-				TargetStatus: &ts,
-			},
-			SecretDataGetter: &remotesecrets.SecretDataGetter{
-				Storage: f.storage,
-			},
-			ObjectMarker: &namespacetarget.NamespaceObjectMarker{},
+		if ts.Error != "" {
```

Comment:
Could there be an edge case where even if the target has an error, we still need to cleanup? If not, ignore this.

Reply:
I've added a little bit of the explanation here to provide reasons why this should be ok.